be/src/olap/olap_server.cpp

```diff
@@ -39,23 +39,28 @@ namespace doris {
 // number of running SCHEMA-CHANGE threads
 volatile uint32_t g_schema_change_active_threads = 0;
 
+// background threads is detached thread, but they will depend on 'this' pointer.
+// so we should pass a shared_from_this ptr to make sure that the pointer is not destroyed
+// before the thread exits
```

Comment:
In my opinion, we should ensure all threads stoped before destrory the StorageEngine instance, use smart pointer is a compromised way to resolve this problem. There is no reason to keep any thread running after StorageEngine stoped in main function.